admin/metabox/class-metabox.php

```diff
@@ -756,10 +773,80 @@ private function get_replace_vars() {
 			$cached_replacement_vars[ $var ] = wpseo_replace_vars( '%%' . $var . '%%', $post );
 		}
 
-		return $cached_replacement_vars;
+		// Merge custom replace variables with the WordPress ones.
+		return array_merge( $cached_replacement_vars, $this->get_custom_replace_vars( $post ) );
+	}
+
+	/**
+	 * Gets the custom replace variables for custom taxonomies and fields.
+	 *
+	 * @param WP_Post $post The post to check for custom taxonomies and fields.
+	 *
+	 * @return array Array containing all the replacement variables.
+	 */
+	private function get_custom_replace_vars( $post ) {
+		return array(
+			'custom_fields' => $this->get_custom_fields_replace_vars( $post ),
+			'custom_taxonomies' => $this->get_custom_taxonomies_replace_vars( $post ),
+		 );
 	}
 
 	/**
+	 * Gets the custom replace variables for custom taxonomies.
+	 *
+	 * @param WP_Post $post The post to check for custom taxonomies.
+	 *
+	 * @return array Array containing all the replacement variables.
+	 */
+	private function get_custom_taxonomies_replace_vars( $post ) {
+		$taxonomies = get_object_taxonomies( $post, 'objects' );
+		$built_in = get_taxonomies( array( '_builtin' => true, 'public' => true ) );
+
+		$custom_replace_vars = array();
+
+		foreach ( $taxonomies as $taxonomy_name => $taxonomy ) {
+			if ( isset( $built_in[ $taxonomy_name ] ) === true ) {
```

Comment:
There should be a `_builtin` and `public ` available for each taxonomy fetched with `get_object_taxonomies`. If it's so `$build_in` isn't necessary.